### PR TITLE
Optimize node ancestor checks

### PR DESCRIPTION
**Pull request by tvasenin**
Eliminate String.split() and compare position strings directly.

Gives huge performance boost when handling lots of nodes, e.g.
https://react-component.github.io/tree-select/examples/big-data.html

### CODE REVIEW

**Comment by tvasenin** [comment]
The corresponding PR for tree-select: https://github.com/react-component/tree-select/pull/84

File: examples/util.js [hunk 1]
```diff
@@ -37,12 +37,17 @@ console.log('总节点数（单个tree）：', calcTotal());
...
-export function isInclude(smallArray, bigArray) {
-  return smallArray.every((ii, i) => {
-    return ii === bigArray[i];
-  });
+function isPositionPrefix(smallPos, bigPos) {
+  if (bigPos.length < smallPos.length) {
+    return false;
+  }
+  // attention: "0-0-1" "0-0-10"
+  if ((bigPos.length > smallPos.length) && (bigPos.charAt(smallPos.length) !== '-')) {
```
Comment (yesmeck) [inline examples/util.js]: https://jsperf.com/indexof-vs-charat

Seems like `indexOf` faster than `charAt`.

Comment (yesmeck) [reply examples/util.js]: Nevermind.

**Comment by benjycui** [comment]
这覆盖率跌得也是诡异。

**Comment by tvasenin** [comment]
@yesmeck Please, take a look at the similar commit in tree-select repo: https://github.com/react-component/tree-select/pull/84

**Comment by afc163** [comment]
Great job!  @tvasenin

**Pull request merged**